### ISSUE TITLE
Event Hubs lose connection with clients

### ISSUE THREAD

**Issue by Omzig** [ironmansoftware/powershell-universal#4003]
### Description of Issue

The Event Hub clients will lose their connection at some point....
- setup a server
- setup 2 clients
- configure the 2 clients and enable auth with a token
- Get everything hooked up and run a script to send a message to all the clients.
- Verify it works

Problem:
- wake up the next morning
- run the job that sends the message to the clients
```
[error] Connection 'LG6oOFcGoBdnv2Sq6r2Piw' does not exist. 
[error] Connection 'i0F7FSftflmcu3rBPBNGZA' does not exist. 
```

Workaround:
- Restart the agents
- test again, if it fails, restart the app pool or the psu server service
- restart agents
- test again, it might work

Other Notes:
- sometimes a live connection will be listed with no disconnect date, and you cannot communicate to a connection that is lost, but the server thinks it is still active.

### Version

4.4.1

### Severity

Medium

### Hosting Method

MSI (Windows Service)

### Operating System

Windows

### Database

LiteDB

### Licensed

Yes

### Features

Event Hubs
Event Hub Client

**Comment by adamdriscoll** [comment]
Do the logs on the event hub agent have any indication that it is disconnecting?

**Comment by RockoTheeHut** [comment]
This is really odd timing... I just found that when I restarted PSU that the agents never reconnected. The service was still running, but that's it. 

The log states that it lost connection but there is no indication in the log that it attempted to reconnect. Restarting the agent established an agent connection again. 

2024-10-31 15:37:32.294 -04:00 [WRN] Connection to server https://bob.com at hub BOB_Dev_Cmd closed.
2024-10-31 15:37:32.294 -04:00 [WRN] Connection to server https://bob.com at hub BOB_Dev closed.

**Comment by adamdriscoll** [comment]
I just realized this is for v4. I implemented a retry in v5 but it looks like it was never merged into v4. Easy to do. 

Edit to myself: We should probably do some sort of heartbeat call down the event hub client because, if the server doesn't see it disconnecting, then we will run into other issues. The heartbeat could check to see if we still have a connection and set it to disconnected if not. We can just add it to the heartbeat job in the server.
